php-checks/src/test/resources/checks/CountInsteadOfEmptyCheck.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+if (count($a) >= 0) { // Noncompliant
+//  ^^^^^^^^^
+  echo $a[0];
+}
+
+if (count($a_2) >= 0) { // Compliant
+  echo $a_2;
+}
+
+if (0 < count($a_3)) { // Noncompliant
+  echo $a_3[0];
+}
+
+if (count($a_4) >= 1) { // Compliant
```

Comment:
My initial thought there was that it is not a check for emptiness and thus irrelevant. But seeing it now, it is a check for emptiness (similar to count($x) == 1) and can be replaced with !empty($x) . Thus, it is currently a FN. If it would be "if (count($a_4) >= 2) ", it would reflect what I initially wanted to test there, because it cannot be replaced with empty() .

Reply:
Fixed. It is detected now. In fact, I've found a good approach to check if it is an equivalent to a usage of empty() in implementation for the Java analyzer and adopted it.